python-client/giskard/core/suite.py

```diff
@@ -40,25 +40,49 @@
 ]
 
 
-class TestSuiteResult(tuple):
+def _param_to_str(param: Any) -> str:
+    if issubclass(type(param), Dataset) or issubclass(type(param), BaseModel):
+        return param.name if getattr(param, "name", None) is not None else str(param.id)
+
+    if issubclass(type(param), Artifact):
+        return param.meta.name if param.meta.name is not None else str(param.meta.uuid)
+
+    return str(param)
+
+
+def _params_to_str(params: Dict[str, Any]) -> Dict[str, str]:
+    return {key: _param_to_str(value) for key, value in params.items()}
+
+
+class TestSuiteResult:
     """Represents the result of a test suite."""
 
+    def __init__(self, passed: bool, results: List[Tuple[str, TestResult, Dict[str, Any]]]):
+        self.passed = passed
+        self.results = results
+
+    def __repr__(self):
+        return f"<TestSuiteResult ({'passed' if self.passed else 'failed'})>"
+
     def _repr_html_(self):
-        passed = self[0]
-        tests_results = "".join(
-            [
-                f"<h3>Test: {key}</h3>{(TestResult(passed=value) if type(value) == bool else value)._repr_html_()}"
-                for key, value in self[1]
-            ]
+        from jinja2 import Environment, PackageLoader, select_autoescape
+        from giskard.visualization.custom_jinja import pluralize, format_metric
+
+        env = Environment(
+            loader=PackageLoader("giskard.visualization", "templates"),
+            autoescape=select_autoescape(),
         )
```

Comment:
I would actually move all of this in the `viz` package, and here do only something like:

```
widget = TestSuiteWidget(self, **other_params)  # TestSuiteWidget(suite: TestSuite, …)
return widget.render_html()
```
So that we decouple jinja and the things related to the rendering, which do not belong to the `TestSuite` internals.